src/core/client/stream/tabs/Configure/Q&A/ExpertSelectionContainer.tsx

```diff
@@ -0,0 +1,353 @@
+import { Localized } from "@fluent/react/compat";
+import React, {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useState,
+} from "react";
+import { RelayPaginationProp } from "react-relay";
+
+import {
+  graphql,
+  useLoadMore,
+  useMutation,
+  useRefetch,
+  withPaginationContainer,
+} from "coral-framework/lib/relay";
+import { GQLUSER_ROLE_RL, GQLUSER_STATUS_RL } from "coral-framework/schema";
+import { Button, Flex, Icon, Typography } from "coral-ui/components";
+import { TextField as SearchTextField } from "coral-ui/components/v2";
+
+import { ExpertSelectionContainer_query as QueryData } from "coral-stream/__generated__/ExpertSelectionContainer_query.graphql";
+import { ExpertSelectionContainerPaginationQueryVariables } from "coral-stream/__generated__/ExpertSelectionContainerPaginationQuery.graphql";
+
+import AddExpertMutation from "./AddExpertMutation";
+import ExpertListItem from "./ExpertListItem";
+import ExpertSearchList from "./ExpertSearchList";
+import RemoveExpertMutation from "./RemoveExpertMutation";
+
+import styles from "./ExpertSelectionContainer.css";
+
+interface Props {
+  storyID: string;
+  query: QueryData | null;
+  relay: RelayPaginationProp;
+}
+
+function computeUsers(query: QueryData | null) {
+  if (!query) {
+    return [];
+  }
+
+  return query.users.edges.map(edge => edge.node);
+}
+
+function computeExperts(query: QueryData | null) {
+  if (!query) {
+    return [];
+  }
+  if (!query.story) {
+    return [];
+  }
+
+  return query.story.settings.experts;
+}
+
+const ExpertSelectionContainer: FunctionComponent<Props> = ({
+  storyID,
+  query,
+  relay,
+}) => {
+  const users = computeUsers(query);
+  const experts = computeExperts(query);
+
+  const searchFieldRef = React.createRef<HTMLInputElement>();
+  const searchRootRef = React.createRef<HTMLDivElement>();
+
+  const [loadMore, isLoadingMore] = useLoadMore(relay, 10);
+  const [searchFilter, setSearchFilter] = useState<string>("");
+  const [roleFilter] = useState<GQLUSER_ROLE_RL | null>(null);
+  const [statusFilter] = useState<GQLUSER_STATUS_RL | null>(null);
+  const [, isRefetching] = useRefetch(relay, {
+    searchFilter: searchFilter || null,
+    roleFilter,
+    statusFilter,
+  });
+  const [tempSearchFilter, setTempSearchFilter] = useState<string>("");
+  const loading = !query || isRefetching;
+  const hasMore = !isRefetching && relay.hasMore();
+
+  const addExpertMutation = useMutation(AddExpertMutation);
+  const removeExpertMutation = useMutation(RemoveExpertMutation);
+
+  const clearSearchFilter = useCallback(() => {
+    const searchInputNode: any = searchFieldRef.current;
+    searchInputNode.value = "";
```

Comment:
can we manage the field state in a more react-y way?

Reply:
I thought ref's were part of React? 😆 
I get what you mean though, I'll try and find a cleaner way to clear this field.